Application/EdFi.Security.DataAccess/Repositories/SecurityRepository.cs

```diff
@@ -72,14 +70,21 @@ private IEnumerable<ResourceClaim> GetResourceClaimLineageForResourceClaim(strin
                 throw new InvalidOperationException($"Multiple resource claims with a claim name of '{resourceClaimUri}' were found in the Ed-Fi API's security configuration. Authorization cannot be performed.", ex);
             }
 
-            if (resourceClaim != null)
+            if (resourceClaim == null)
+            {
+                return resourceClaimLineage;
+            }
+            
+            if (resourceClaimLineage.Select(rc => rc.ClaimName).Contains(resourceClaim.ClaimName))
```

Comment:
Optional: This could also be expressed as:
```suggestion
            if (resourceClaimLineage.Any(rc => rc.ClaimName == resourceClaim.ClaimName))
```